Engine/AlgorithmManager.cs

```diff
@@ -262,6 +263,12 @@ public void Run(AlgorithmNodePacket job, IAlgorithm algorithm, IDataFeed feed, I
                             results.SamplePerformance(_previousTime.Date, Math.Round((algorithm.Portfolio.TotalPortfolioValue - portfolioValue) * 100 / portfolioValue, 10));
                         }
                         portfolioValue = algorithm.Portfolio.TotalPortfolioValue;
+
+                        if (portfolioValue <= 0)
+                        {
+                            Log.Trace("AlgorithmManager.Run(): Portfolio value is less than or equal to zero");
```

Comment:
This won't pass the error message outside of the console (it's a silent death). We should pass it out via a algorithm log or message as well.  e.g. result.SystemDebugMessage();